drivers/dac/dac_mcp4725.c

```diff
@@ -0,0 +1,158 @@
+/*
+ * Copyright (c) 2021 Laird Connectivity
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#define DT_DRV_COMPAT microchip_mcp4725
+
+#include <zephyr.h>
+#include <drivers/i2c.h>
+#include <drivers/dac.h>
+#include <logging/log.h>
+
+LOG_MODULE_REGISTER(dac_mcp4725, CONFIG_DAC_LOG_LEVEL);
+
+/* Information in this file comes from MCP4725 datasheet revision D
+ * found at https://ww1.microchip.com/downloads/en/DeviceDoc/22039d.pdf
+ */
+
+/* Defines for field values in MCP4725 DAC register */
+#define MCP4725_DAC_MAX_VAL			0xFFF
+
+#define MCP4725_FAST_MODE_POWER_DOWN_POS	4U
+#define MCP4725_FAST_MODE_DAC_UPPER_VAL_POS	8U
+#define MCP4725_FAST_MODE_DAC_UPPER_VAL_MASK	0xF
+#define MCP4725_FAST_MODE_DAC_LOWER_VAL_MASK	0xFF
+
+#define MCP4725_READ_RDY_POS			7U
+#define MCP4725_READ_RDY_MASK			(0x1 << MCP4725_READ_RDY_POS)
+
+/* After writing eeprom, the MCP4725 can be in a busy state for 25 - 50ms
+ * See section 1.0 of MCP4725 datasheet, 'Electrical Characteristics'
+ */
+#define MCP4725_BUSY_TIMEOUT_MS			60U
+
+struct mcp4725_config {
+	const char *i2c_master_dev_name;
+	uint16_t i2c_slave_addr;
+};
+
+struct mcp4725_data {
+	const struct device *i2c_master;
+};
```

Comment:
The terms master & slave [should not be used](https://docs.zephyrproject.org/latest/guides/coding_guidelines/index.html#rule-a-2-inclusive-language) anymore. In addition to that I suggest to switch to new (more efficient) `DEVICE_DT_GET` API to obtain the device (see below). The `_data` struct is not needed anymore afterwards.
```suggestion
struct mcp4725_config {
	const struct device *i2c_dev;
	uint16_t i2c_addr;
};
```

Reply:
Change made